src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/AsType.cs

```diff
@@ -141,6 +141,53 @@ protected override bool IsValidAsyncOrImpureNode(TexlNode node, TexlBinding bind
             // AsType should always be marked as valid regardless of it being async and impure.
             return true;
         }
+    }
+
+    // AsType(UntypedObject:P, Type:*[]): ![]
+    internal class AsTypeUOFunction : BuiltinFunction
+    {
+        public const string AsTypeInvariantFunctionName = "AsType";
+
+        public override bool IsSelfContained => true;
+
+        public override bool SupportsParamCoercion => false;
+
+        public override bool IsRestrictedUDFName => true;
+
+        public override bool HasTypeArgs => true;
+
+        public override bool ArgIsType(int argIndex)
+        {
+            return argIndex == 1;
+        }
+
+        public AsTypeUOFunction()
+            : base(AsTypeInvariantFunctionName, TexlStrings.AboutAsTypeUO, FunctionCategories.REST, DType.Error, 0, 2, 2, DType.UntypedObject, DType.Error)
+        {
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.AsTypeUOArg1, TexlStrings.AsTypeUOArg2 };
+        }
+
+        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == 2);
+            Contracts.Assert(argTypes.Length == 2);
+            Contracts.AssertValue(errors);
+
+            if (!base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap))
+            {
+                return false;
```

Comment:
Curious: would something like `AsType(ParseJSON("123"), 1)` succeed (and return the number 123)? I don't know if CheckTypes would handle a non-type argument properly. Whatever the answer, can we have a test for that? :-)

Reply:
it would fail.  
because of this - https://github.com/microsoft/Power-Fx/pull/2569/files#diff-3b97f603fa1daea008a12cd0138ee5a6bf8270bc878af497c2ff4a272d04a583R5128

somewhat equivalent test is this https://github.com/microsoft/Power-Fx/pull/2569/files#diff-a2864451c36106fe453a61cef0229cc6b6dd73c0eb77ad88485dd954fbf5a804R123 .

will add the suggested test too